apps/air-discount-scheme/web/screens/Subsidy/components/Benefits/Benefits.tsx

```diff
@@ -71,7 +71,7 @@ function Benefits({ misc }: PropTypes) {
     copySuccess,
     connectionFlightInfo,
   } = JSON.parse(misc)
-  const benefits = discounts.filter(({ user }) => user.meetsADSRequirements)
+  const benefits = discounts
```

Comment:
This looks scary but the backend has been made responsible for filtering since this was first written

Reply:
Then maybe add a in-code comment explaining this?

Reply:
I think the assumption should be that the backend doesn't just give you codes when you don't have the rights.
Why should a discountCode be generated for a user who doesn't meet requirements?
IMO the filter was from a time where the backend didn't perform the checks it should have.

Reply:
Ah yes sorry, the big scary was about what I was removing, not what is remaining.
I should've been more clear.

There was a protective filter that I removed (scary).
This filter is no longer needed :)